mavros/src/plugins/setpoint_trajectory.cpp

```diff
@@ -0,0 +1,165 @@
+/**
+ * @brief SetpointTRAJECTORY plugin
+ * @file setpoint_trajectory.cpp
+ * @author Jaeyoung Lim <jaeyoung@auterion.com>
+ *
+ * @addtogroup plugin
+ * @{
+ */
+/*
+ * Copyright 2019 Jaeyoung Lim.
+ *
+ * This file is part of the mavros package and subject to the license terms
+ * in the top-level LICENSE file of the mavros repository.
+ * https://github.com/mavlink/mavros/tree/master/LICENSE.md
+ */
+
+#include <mavros/mavros_plugin.h>
+#include <mavros/setpoint_mixin.h>
+#include <eigen_conversions/eigen_msg.h>
+
+#include <nav_msgs/Path.h>
+#include <mavros_msgs/PositionTarget.h>
+#include <trajectory_msgs/MultiDOFJointTrajectory.h>
+#include <trajectory_msgs/MultiDOFJointTrajectoryPoint.h>
+
+namespace mavros {
+namespace std_plugins {
+/**
+ * @brief Setpoint TRAJECTORY plugin
+ *
+ * Receive trajectory setpoints and send setpoint_raw setpoints along the trajectory.
+ */
+class SetpointTrajectoryPlugin : public plugin::PluginBase,
+	private plugin::SetPositionTargetLocalNEDMixin<SetpointTrajectoryPlugin> {
+public:
+	SetpointTrajectoryPlugin() : PluginBase(),
+		sp_nh("~setpoint_trajectory")
+	{ }
+
+	void initialize(UAS &uas_)
+	{
+		PluginBase::initialize(uas_);
+
+		sp_nh.param<std::string>("frame_id", frame_id, "map");
```

Comment:
Do we want this to be `map` or `odom` by default?

Reply:
This I don't have a good insight. What would be a better way to handle this?

Reply:
Depends. Are we navigating within an arbitrary navigation frame or a well defined global frame (resulting from GPS coordinates for example)?

Reply:
I guess it is safe to assume we are navigating in the local frame for now. Would `map` be an appropriate frame for that?

Reply:
Maybe this helps you take thag decision: https://www.ros.org/reps/rep-0105.html

Reply:
Also, if we are sending setpoints exclusively in the local frame, then that needs to be clarified as a comment. Though, I don't see a reason why we shouldn't send setpoints in the body frame as well, similar to what we do on the other setpoint plugins.

Reply:
How can we handle this with the multiDOFJointTrajectory message? Do you have any good pointers? For me it seems like you just need to make sure you set the correct frame_id

Reply:
The setpoint message has a frame field. You hard setting it to the local frame.

Reply:
https://mavlink.io/en/messages/common.html#SET_POSITION_TARGET_LOCAL_NED

Reply:
Would setting the default to `MAV_FRAME_LOCAL_NED` make sense?

Reply:
No. Check what's being done here: https://github.com/mavlink/mavros/blob/master/mavros/src/plugins/setpoint_position.cpp#L139

Reply:
@Jaeyoung-Lim any news here?

Reply:
Sorry for the delay. I was working on this and was experiencing some problems with the transforms. I will try and get this done this week.